### PR TITLE
Alerting: Add notification policy flow chart

### LINKED ISSUE #45479: Alerting: improve onboarding by adding root route to alert creation flow
### Problem
Notification policies are confusing because labels as a concept aren’t explained. Users in usability tests with no Alerting experience didn't know what would happen to their notifications or what they needed to do to get a working notification. They also didn't understand the concept of labels and how it is necessary for notifications management.

2022-02 Research Outcomes_ Alert creation (2)

### Solution
Display the default notification policy on the alert creation flow; this will make it a bit clearer to the user that there is already a default policy for alert instances.

2022-02 Research Outcomes_ Alert creation (3)

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Adds a flow chart to  a new section in the alert rule creating that will explain how notification policies work. 

Also small change to the theme creation where the name was always set to `Dark`.

**Which issue(s) this PR fixes**:
Fixes #45479

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+export const NotificationsStep: FC = () => {
+  const [hideFlowChart, setHideFlowChart] = useState(false);
+  const styles = useStyles2(getStyles);
+  const theme = useTheme2();
+
+  return (
+    <RuleEditorSection
+      stepNo={4}
+      title="Notifications"
+      description="Grafana handles the notifications for alerts by assigning labels to alerts. These labels connect alerts to contact points and silence alert instances that have matching labels."
```
Comment (joeblubaugh) [inline public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx:18]: nit: I think this mixes some things together - labels are assigned to alerts, and then those labelled alerts go through several sets of rules to decide what gets sent to the contact points.
1. Where do I send these alerts?
2. Are the alerts grouped so only a single alert is sent?
3. Are there any silences or mute timings that stop these alerts from being sent?

These questions are answered by both notification policies and silences - the labels don't "do" anything.

Here's a suggestion, but feel free to stick with what's here already.
> Grafana assigns labels to alerts based on the results of the query and the configuration of the alerting rule. These labels are compared with notification rules and silences to select the correct contact point and determine if an alert has been silenced and should not be sent.

**Review by konrad147 — APPROVED** [review]
LGTM

**Review by gillesdemey — APPROVED** [review]
LGTM, I agree with Joe about the copy but don’t think it’s a blocker

File: public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx (hunk 1 above)
Comment (peterholmberg) [reply public/app/features/alerting/unified/components/rule-editor/NotificationsStep.tsx:18]: @joeblubaugh it's a bit long for a description. I will merge as is and we can figure something out in beta.

**Comment by konrad147** [comment]
Hey Denis,
This should be fixed in 9.0.2 thanks to https://github.com/grafana/grafana/pull/51409

**Pull request merged**